src/linter.coffee

```diff
@@ -1,38 +1,44 @@
-_ = require 'lodash'
-async = require 'async'
-asyncHandlers = require 'async-handlers'
-fsExtra = require 'fs-extra'
-DependencyLinter = require './linter/dependency_linter'
-UsedModuleFinder = require './linter/used_module_finder'
-path = require 'path'
-
-
 class Linter
 
 
-  constructor: (@dir, {allowUnused, devFilePatterns, devScripts, ignoreFilePatterns}) ->
-    @dependencyLinter = new DependencyLinter {allowUnused, devFilePatterns, devScripts}
-    @usedModuleFinder = new UsedModuleFinder {@dir, ignoreFilePatterns}
+  constructor: (@dir, {@allowUnused, @devFilePatterns, @devScripts, @ignoreFilePatterns}) ->
 
 
-  lint: (done) ->
-    async.parallel {
-      listedModules: @getListedModules
-      usedModules: @usedModuleFinder.find
-    }, asyncHandlers.transform(@dependencyLinter.lint, done)
+  lint: (done) =>
+    async = require 'async'
+    asyncHandlers = require 'async-handlers'
+    async.auto {
+      packageJson: @getPackageJson
+      usedModules: @getUsedModules
+    }, asyncHandlers.transform(@lintModules, done)
 
 
   extractListedModules: (packageJson) ->
+    _ = require 'lodash'
```

Comment:
Another observation: This method now does more than one thing. Before, it was simply massaging some data. Now it imports a library, and then uses that library to massage some data. Following refactoring rules, I'd say this should be extracted.